### PR TITLE
Prepare 2024.4.1 release

### PR DESCRIPTION
**Pull request by marcelovilla**
## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [x] Other (please describe): release

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

**Review by dcmcand — APPROVED** [review]

File: src/_nebari/upgrade.py [hunk 1]
```diff
@@ -744,7 +749,27 @@ class Upgrade_2024_4_1(UpgradeStep):
...
+                    default_node_groups = provider_enum_default_node_groups_map[
+                        provider
+                    ]
+                    continue_ = Prompt.ask(
+                        f"Would you like to include the default configuration for the node groups in [purple]{config_filename}[/purple]?",
+                        choices=["y", "N"],
+                        default="N",
+                    )
+                    if continue_ == "y":
+                        config[provider_full_name]["node_groups"] = default_node_groups
```
Comment (Adam-D-Lewis) [inline src/_nebari/upgrade.py:770]: Have you checked if this value is actually written to the nebari-config.yaml file?  If so, then LGTM!

**Review by Adam-D-Lewis — APPROVED** [review]

File: src/_nebari/upgrade.py (hunk 1 above)
Comment (viniciusdc) [reply src/_nebari/upgrade.py:770]: good point, they are matching: 
https://github.com/nebari-dev/nebari/blob/79b1218cb6cec95a9bbb96ad27889015a47d27d8/src/_nebari/stages/infrastructure/__init__.py#L536-L543

https://github.com/nebari-dev/nebari/blob/79b1218cb6cec95a9bbb96ad27889015a47d27d8/src/_nebari/stages/infrastructure/__init__.py#L558-L563

**Review by viniciusdc — APPROVED** [review]
LGTM!

File: src/_nebari/upgrade.py (hunk 1 above)
Comment (marcelovilla) [reply src/_nebari/upgrade.py:770]: @Adam-D-Lewis yes, the config is properly updated when accepting the changes.

@viniciusdc I'm not entirely sure what you mean here.

**Pull request merged**